field/composite.go

```diff
@@ -113,11 +118,35 @@ func (f *Composite) getSubfields() map[string]Field {
 // should only pass None or nil values for ths type. Passing any other value
 // will result in a panic.
 func (f *Composite) SetSpec(spec *Spec) {
```

Comment:
Can we return an error here instead of the panics? It breaks the public API, but returning valid errors is more important to me.

Reply:
We initially thought on that, but the problem with this is that we have to change the field.Field interface, this means that we will have to change all the implementations for this interface, what do you think?

Reply:
I have checked, and it is not an easy change to make because `SetSpec` is being used somewhere along the call chain by `func NewMessage(spec *MessageSpec) *Message` - and I don't know how big would be the breaking change if we add `error` to `NewMessage`.